MekHQ/src/mekhq/campaign/personnel/Person.java

```diff
@@ -1080,123 +894,70 @@ public String getPrimaryRoleDesc() {
         if (isClanner()) {
             bgPrefix = getPhenotype().getShortName() + " ";
         }
-        return bgPrefix + getRoleDesc(getPrimaryRole(), isClanner());
+        return bgPrefix + getPrimaryRole().getName(isClanner());
     }
 
     public String getSecondaryRoleDesc() {
-        return getRoleDesc(getSecondaryRole(), isClanner());
-    }
-
-    public static int getRoleMnemonic(int type) {
-        // The following characters are unused:
-        // J, K, Q, X, Z
-        switch (type) {
-            case T_MECHWARRIOR:
-                return KeyEvent.VK_M;
-            case T_GVEE_DRIVER:
-                return KeyEvent.VK_V;
-            case T_NVEE_DRIVER:
-                return KeyEvent.VK_N;
-            case T_VEE_GUNNER:
-                return KeyEvent.VK_G;
-            case T_AERO_PILOT:
-                return KeyEvent.VK_A;
-            case T_PROTO_PILOT:
-                return KeyEvent.VK_P;
-            case T_CONV_PILOT:
-                return KeyEvent.VK_F;
-            case T_BA:
-                return KeyEvent.VK_B;
-            case T_INFANTRY:
-                return KeyEvent.VK_S;
-            case T_SPACE_PILOT:
-                return KeyEvent.VK_I;
-            case T_SPACE_CREW:
-                return KeyEvent.VK_W;
-            case T_SPACE_GUNNER:
-                return KeyEvent.VK_U;
-            case T_NAVIGATOR:
-                return KeyEvent.VK_Y;
-            case T_MECH_TECH:
-                return KeyEvent.VK_T;
-            case T_MECHANIC:
-                return KeyEvent.VK_E;
-            case T_AERO_TECH:
-                return KeyEvent.VK_O;
-            case T_DOCTOR:
-                return KeyEvent.VK_D;
-            case T_ADMIN_COM:
-                return KeyEvent.VK_C;
-            case T_ADMIN_LOG:
-                return KeyEvent.VK_L;
-            case T_ADMIN_TRA:
-                return KeyEvent.VK_R;
-            case T_ADMIN_HR:
-                return KeyEvent.VK_H;
-            case T_VTOL_PILOT:
-            case T_BA_TECH:
-            case T_ASTECH:
-            case T_MEDIC:
-            case T_LAM_PILOT:
-            case T_VEHICLE_CREW:
-            case T_NONE:
-            default:
-                return KeyEvent.VK_UNDEFINED;
-        }
+        return getSecondaryRole().getName(isClanner());
     }
 
-    public boolean canPerformRole(int role) {
+    public boolean canPerformRole(PersonnelRole role) {
```

Comment:
I'm not certain the logic for this method should stay here or be moved to the PersonnelRole Enum given how much it relies on the latter, but the heavy reliance on Personnel is why I've kept it here.